examples/aws/terraform/xray.tf

```diff
@@ -1,17 +1,17 @@
 # Substation Lambdas use X-Ray for performance monitoring and tuning. If the AWS account's X-Ray data is not encrypted, then use this file to setup encryption. The ARN produced by `xray_key` (or the ARN of a previously created encryption key) must be added as a kms_read and kms_write resource on all Lambda IAM policies. Alternatively, if no encryption is wanted, then exclude this file and any references to the `xray_key` ARN from Lambda IAM policies.
 
-resource "aws_kms_key" "xray_key" {
-  description         = "KMS used for server-side encryption of X-Ray data"
-  enable_key_rotation = true
-}
+# resource "aws_kms_key" "xray_key" {
```

Comment:
i'd be fine with this if we can also make it work with the permissions below, otherwise it's just an example and people can figure it out on their own.

- https://github.com/brexhq/substation/blob/jshlbrd/break/public-api/examples/aws/terraform/example_kms.tf#L10
- https://github.com/brexhq/substation/blob/jshlbrd/break/public-api/examples/aws/terraform/example_kms.tf#L41

Reply:
We could make those work by changing the local to being a variable and doing something like this in the KMS file which allows for users to override with terraform CLI flags or environment variables. 
```terraform
locals = {
   kms_resources = var.enable_xray ? [module.kms_substation.arn, aws_kms_key.xray_key.*.arn] : [module.kms_substation.arn]
}

module "iam_kms_read" {
  source = "../../../build/terraform/aws/iam"
  resources = local.kms_resources
}
...
module "iam_kms_write" {
  source = "../../../build/terraform/aws/iam"
  resources = local.kms_resources
}
```

That being said, I think users can figure this out without leaving commented code in the repo so we could also remove this entirely and rely on the new docs for X-Ray information?

Reply:
yeah, let's remove it for now and if we decide to offer modules that implement specific pipeline designs then we may want to bring the idea back with a config like that.